lib/iris/util.py

```diff
@@ -613,8 +613,14 @@ def column_slices_generator(full_slice, ndims):
 
         list_of_slices.append(first_slice)
 
-    data_ndims = max(dimension_mapping.values())
-    if data_ndims is not None:
+    try:
+        data_ndims = max(dim
+                         for dim in dimension_mapping.values()
+                         if dim is not None)
+    except ValueError:
+        # No non-None dimensions
+        data_ndims = None
+    else:
```

Comment:
How about we just ditch this code block? I think we can use `_count_current_dim` instead:
```python
spanning_slice_with_tuple = [slice(None, None)] * _count_current_dim
```